openbas-api/src/main/java/io/openbas/execution/ExecutionExecutorService.java

```diff
@@ -27,30 +29,87 @@ public class ExecutionExecutorService {
+        agents.stream().filter(agent -> agent.getExecutor() == null).toList();
+    agents.removeAll(withoutExecutorAgent);
+    List<Agent> crowdstrikeAgents =
+        agents.stream()
+            .filter(agent -> CROWDSTRIKE_EXECUTOR_TYPE.equals(agent.getExecutor().getType()))
+            .collect(Collectors.toList());
+    agents.removeAll(crowdstrikeAgents);
 
     InjectStatus injectStatus =
         inject.getStatus().orElseThrow(() -> new IllegalArgumentException("Status should exist"));
     AtomicBoolean atLeastOneExecution = new AtomicBoolean(false);
-    AtomicBoolean atOneTraceAdded = new AtomicBoolean(false);
-
+    AtomicBoolean atLeastOneTraceAdded = new AtomicBoolean(false);
+    // Manage inactive agents
+    if (!inactiveAgents.isEmpty()) {
+      inactiveAgents.forEach(
+          agent ->
+              injectStatus.addTrace(
+                  ExecutionTraceStatus.AGENT_INACTIVE,
+                  "Agent "
+                      + agent.getExecutedByUser()
+                      + " is inactive for the asset "
+                      + agent.getAsset().getName(),
+                  ExecutionTraceAction.COMPLETE,
+                  agent));
+      atLeastOneTraceAdded.set(true);
+    }
+    // Manage without executor agents
+    if (!withoutExecutorAgent.isEmpty()) {
+      inactiveAgents.forEach(
+          agent ->
+              injectStatus.addTrace(
+                  ExecutionTraceStatus.ERROR,
+                  "Cannot find the executor for the agent "
+                      + agent.getExecutedByUser()
+                      + " from the asset "
+                      + agent.getAsset().getName(),
+                  ExecutionTraceAction.COMPLETE,
+                  agent));
+      atLeastOneTraceAdded.set(true);
+    }
+    // Manage Crowdstrike agents for batch execution
+    try {
+      ExecutorContextService executorContextService =
+          context.getBean(CROWDSTRIKE_EXECUTOR_NAME, ExecutorContextService.class);
+      crowdstrikeAgents =
+          executorContextService.launchBatchExecutorSubprocess(
+              inject, crowdstrikeAgents, injectStatus);
+      atLeastOneExecution.set(true);
+    } catch (Exception e) {
+      crowdstrikeAgents.forEach(
+          agent ->
+              injectStatus.addTrace(
+                  ExecutionTraceStatus.ERROR,
+                  e.getMessage(),
+                  ExecutionTraceAction.COMPLETE,
+                  agent));
+      atLeastOneTraceAdded.set(true);
+    }
+    // Manage remaining agents
     agents.forEach(
         agent -> {
           try {
             launchExecutorContextForAgent(inject, agent);
             atLeastOneExecution.set(true);
           } catch (AgentException e) {
-            ExecutionTraceStatus traceStatus =
-                e.getMessage().startsWith("Agent error")
-                    ? ExecutionTraceStatus.AGENT_INACTIVE
-                    : ExecutionTraceStatus.ERROR;
             injectStatus.addTrace(
```

Comment:
Should we log the exception in addition to inserting the traces, in order to have the full stack trace somewhere ?